src/test/java/org/folio/edge/rtac/utils/RtacUtilTest.java

```diff
@@ -0,0 +1,81 @@
+package org.folio.edge.rtac.utils;
+
+import static io.vertx.core.http.HttpHeaders.ACCEPT;
+import static org.folio.edge.core.Constants.TEXT_PLAIN;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+
+@RunWith(VertxUnitRunner.class)
+public class RtacUtilTest {
+
+  @Mock
+  private static HttpServerRequest rtacRequest;
+  @Mock
+  private static MultiMap rtacHeaders;
+
+  private static List<String> supportedAcceptHeaders = Arrays
+      .asList(RtacMimeTypeEnum.getAllTypesAsString());
+  private static List<String> unsupportedAcceptHeaders = Arrays.asList(TEXT_PLAIN);
+
+  @BeforeClass
+  public static void beforeClassRtacUtils() {
+    // Creates mock of RTAC request nd headers
+    rtacRequest = mock(HttpServerRequest.class);
+    rtacHeaders = mock(MultiMap.class);
+
+    // Define mock object's behavior
+    when(rtacRequest.headers()).thenReturn(rtacHeaders);
+  }
+
+  @Test
+  public void shouldSupportedTypePassedWhenClientSpecifiedSupportedType() {
+    when(rtacHeaders.contains(ACCEPT)).thenReturn(true);
+    when(rtacHeaders.getAll(ACCEPT)).thenReturn(supportedAcceptHeaders);
+    assertTrue(RtacUtils.checkSupportedAcceptHeaders(rtacRequest));
+  }
+
+  @Test
+  public void shouldCheckingForSupportedTypePassedWhenClientDoesNotStateAPreference() {
+    when(rtacHeaders.contains(ACCEPT)).thenReturn(false);
+    assertTrue(RtacUtils.checkSupportedAcceptHeaders(rtacRequest));
+  }
+
+  @Test
+  public void shouldCheckingForSupportedTypeFailedWhenClientSpecifiedUnsupportedType() {
+    when(rtacHeaders.contains(ACCEPT)).thenReturn(true);
+    when(rtacHeaders.getAll(ACCEPT)).thenReturn(unsupportedAcceptHeaders);
+    assertFalse(RtacUtils.checkSupportedAcceptHeaders(rtacRequest));
+  }
+
+  @Test
+  public void shouldCheckingForXMLTypePassedWhenClientDoesNotStateAPreference() {
+    when(rtacHeaders.contains(ACCEPT)).thenReturn(false);
+    assertTrue(RtacUtils.isXmlRequest(rtacRequest));
+  }
+
+  @Test
+  public void shouldCheckingForXMLTypePassedWhenClientSpecifiedSupportedType() {
```

Comment:
Does XML take precedence over JSON if both media types are supported by the client, or is this a side effect of the order of the types in the enumeration?

Reply:
Yes, XML has higher priority, even if both of types presented.

2. Scenario – return JSON
Given request to edge-rtac
When Accept header in request includes application/json
And application/xml isn’t included in Accept header
Then response is returned in JSON format

Reply:
Ok, what do you think about how this describes this behaviour:

```suggestion
  public void XmlIsPreferredWhenClientAcceptsBothXmlAndJson() {
```

Reply:
Ok, what do you think about how this describes this behaviour:

```suggestion
  public void XmlIsPreferredWhenClientAcceptsBothXmlAndJson() {
```

Reply:
I'm not sure I understand how this test is different to `shouldSupportedTypePassedWhenClientSpecifiedSupportedType`, can you help me understand?

Reply:
Agree there was some kind of duplication in these case. The first one just receive all the headers and checks that verification just passed. Whereas second (which name will be updated) tests if both accepted types received and request determined like XML, so therefore XML payload will be returned (this part of testing covered in the RtacHandlerTest).